src/herbpy/action/rogue.py

```diff
@@ -0,0 +1,354 @@
+    @param manip The manipulator to perform the sweep
+    @param margin The distance between the start object and the hand,
+                  so the vertical space between the hand and objects. 
+                  This must be enough to clear the objects themselves.
+    @param render Render tsr samples during planning
+    """
+    if manip is None:
+        manip = robot.GetActiveManipulator()
+
+    #ee_offset : such that the hand, not wrist, is above the object
+    #hand_pose : places the hand above the start location
+    if manip.GetName() == 'right':
+        hand = robot.right_hand
+        ee_offset = -0.2
+        hand_pose = numpy.array([[ 0, -1, 0, start_coords[0]],
+                                 [ 0,  0, 1, (start_coords[1]+ee_offset)],
+                                 [-1,  0, 0, (start_coords[2]+margin)],
+                                 [ 0,  0, 0, 1]])
+
+    elif manip.GetName() == 'left':
+        hand = robot.left_hand
+        ee_offset = 0.2
+        hand_pose = numpy.array([[ 0,  1, 0, start_coords[0]],
+                                 [ 0,  0, -1, (start_coords[1]+ee_offset)],
+                                 [-1,  0, 0, (start_coords[2]+margin)],
+                                 [ 0,  0, 0, 1]])
+  
+    else:
+        raise prpy.exceptions.PrPyException('Manipulator does not have an \
+                 associated hand')
+
+    end_trans = numpy.eye(4)
+    end_trans[0:3, 3] = end_coords
+
+    hand.MoveHand(f1=1, f2=1, f3=1, spread=3.14)
+    manip.PlanToEndEffectorPose(hand_pose)
+
+    #TSR to sweep to end position
+    sweep_tsr = robot.tsrlibrary(None, 'sweep', end_trans, manip)
+
+    p = openravepy.KinBody.SaveParameters
+    with robot.CreateRobotStateSaver(p.ActiveManipulator):
+        robot.SetActiveManipulator(manip)
+        with prpy.viz.RenderTSRList(sweep_tsr, robot.GetEnv(), render=render):
+             robot.PlanToTSR(sweep_tsr, execute=True)
+
+@ActionMethod
+def Exhibit(robot, obj, manip=None, distance=0.1, wait=2, release=True, render=True):
+    """
+    @param robot The robot performing the exhibit
+    @param obj The object being exhibited
+    @param manip The maniplator to perform the exhibit
+    @param distance The distance the object will be lifted up
+    @param wait The amount of time the object will be held up in seconds
+    @param render Render tsr samples during planning
+    """
+    with robot.GetEnv():
+        if manip is None:
+            manip = robot.GetActiveManipulator()
+        preconfig = manip.GetDOFValues()
+        robot.Grasp(obj)
+
+    #Lift the object - write more tsrs
+    lift_tsr = robot.tsrlibrary(obj, 'lift', manip, distance=distance)
+    
+    with prpy.viz.RenderTSRList(lift_tsr, robot.GetEnv(), render=render):
+        robot.PlanToTSR(lift_tsr, execute=True)
+
+    #Wait for 'time'
+    time.sleep(wait)
+
+    #'Unlift' the object, so place it back down
+    unlift_tsr = robot.tsrlibrary(obj, 'lift', manip, distance=-distance)
```

Comment:
I don't know if negative `distance` values are supported here.

@jeking04 Does this make sense?